### PR TITLE
New lib: auth-nest-tools

### PR DESCRIPTION
**Pull request by arnifreyrs**
# \<Description\>

Extract nest decorators, guards and modules from auth-api-lib into a new typescript strict library that is smaller and easier to import into other projects.

## Why

auth-api-lib was perhaps too big of a library, which was not typescript strict so it was harder to use in other projects. This new lib is way smaller, has dedicated purpose, and is typescript strict.

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by valurefugl — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -70,3 +70,6 @@
 
 /apps/skilavottord/                                       @island-is/deloitte
 /libs/skilavottord/                                       @island-is/deloitte
+
+# /libs/auth-api-lib/                                     @island-is/fuglar
```
Comment (jeremybarbet) [inline .github/CODEOWNERS:74]: Why is it commented?

**Review by jeremybarbet — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (arnifreyrs) [reply .github/CODEOWNERS:74]: I got weird bugs in the build that the team had no permission or something. I will uncomment this in the next PR when the new lib is part of main

**Review by Thorrik58 — APPROVED** [review]
LGTM

**Review by obmagnusson — APPROVED** [review]
LGTM

**Review by dabbeg — APPROVED** [review]

**Review by pshomov — APPROVED** [review]
LGTM

**Pull request merged**